reporting/impl/src/main/java/org/jboss/windup/reporting/rules/generation/techreport/GetNumberOfLibrariesMethod.java

```diff
@@ -0,0 +1,87 @@
+package org.jboss.windup.reporting.rules.generation.techreport;
+
+import freemarker.ext.beans.StringModel;
+import freemarker.template.TemplateModelException;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.ArchiveModel;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.reporting.freemarker.WindupFreeMarkerMethod;
+import org.jboss.windup.util.ExecutionStatistics;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Gets the count of library occurences recursively per input application.
+ *
+ * <p> Called from a freemarker template as follows:
+ *
+ * <pre>
+ *      getNumberOfLibraries( projectToCount: ProjectModel ): Integer
+ * </pre>
+ *
+ * <p> Returns an Integer object, which holds the count:
+
+ *
+ * @author <a href="mailto:mbrophy@redhat.com">Mark Brophy</a>
+ */
+public class GetNumberOfLibrariesMethod implements WindupFreeMarkerMethod {
+    public static final Logger LOG = Logger.getLogger(GetNumberOfLibrariesMethod.class.getName());
+    private static final String NAME = "getNumberOfLibraries";
+
+    private GraphContext graphContext;
+
+    @Override
+    public void setContext(GraphRewrite event)
+    {
+        this.graphContext = event.getGraphContext();
+    }
+
+    @Override
+    public String getMethodName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Takes a " + ProjectModel.class.getSimpleName()
+                + " as a parameter and returns a count of the number of libraries contained within";
+    }
+
+    @Override
+    public Object exec(@SuppressWarnings("rawtypes") List arguments) throws TemplateModelException
+    {
+        ExecutionStatistics.get().begin(NAME);
+
+        // Function arguments
+        ProjectModel projectModel = null;
+
+        // The project. May be null -> all input applications.
```

Comment:
I'm pretty sure this line is wrong. Won't null just trigger a NullPointerException?

It might be better to check for null and throw a windup exception with a detailed message.